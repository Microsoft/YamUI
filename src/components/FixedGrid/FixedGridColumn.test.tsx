/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FixedGridColumn, FixedGridColumnProps } from '.';

describe('<FixedGridColumn />', () => {
  let component: ShallowWrapper<FixedGridColumnProps>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<FixedGridColumn>grid content</FixedGridColumn>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<FixedGridColumn className="TEST_CLASSNAME">grid content</FixedGridColumn>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when set to fixed', () => {
    beforeEach(() => {
      component = shallow(<FixedGridColumn fixed={true}>grid content</FixedGridColumn>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when set to fixed and given a width', () => {
    beforeEach(() => {
      component = shallow(
        <FixedGridColumn fixed={true} width={150}>
          grid content
        </FixedGridColumn>,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when given a width but not fixed', () => {
    beforeEach(() => {
      component = shallow(<FixedGridColumn width={150}>grid content</FixedGridColumn>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with vertical align middle', () => {
    beforeEach(() => {
      component = shallow(<FixedGridColumn verticalAlign="middle">grid content</FixedGridColumn>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
