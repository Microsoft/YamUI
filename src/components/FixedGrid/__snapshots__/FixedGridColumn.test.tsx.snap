// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FixedGridColumn /> when given a width but not fixed matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn y-fixedGridColumn__hasWidth"
  style={Object {}}
>
  grid content
</div>
`;

exports[`<FixedGridColumn /> when set to fixed and given a width matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn y-fixedGridColumn__isFixed y-fixedGridColumn__hasWidth"
  style={
    Object {
      "width": "150px",
    }
  }
>
  grid content
</div>
`;

exports[`<FixedGridColumn /> when set to fixed matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn y-fixedGridColumn__isFixed"
  style={Object {}}
>
  grid content
</div>
`;

exports[`<FixedGridColumn /> with additional className matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn TEST_CLASSNAME"
  style={Object {}}
>
  grid content
</div>
`;

exports[`<FixedGridColumn /> with default options matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn"
  style={Object {}}
>
  grid content
</div>
`;

exports[`<FixedGridColumn /> with vertical align middle matches its snapshot 1`] = `
<div
  className="y-fixedGridColumn"
  style={Object {}}
>
  <div
    className="y-fixedGridColumn--inner__middle"
  >
    grid content
  </div>
</div>
`;
